File: solutions/elixir/log-level/1/lib/log_level.ex
defmodule LogLevel do
  def to_label(level, legacy?) do
    cond do
      level == 0 && !legacy? -> :trace
      level == 0 && legacy? -> :unknown
      level == 1 -> :debug
      level == 2 -> :info
      level == 3 -> :warning
      level == 4 -> :error
      level == 5 && !legacy? -> :fatal
      level == 5 && legacy? -> :unknown
      true -> :unknown
    end
  end

  def alert_recipient(level, legacy?) do
    log_level = to_label(level, legacy?)
    cond do
      log_level == :error || log_level == :fatal -> :ops
      log_level == :unknown && legacy? -> :dev1
      log_level == :unknown -> :dev2
      true -> false
    end
  end
end

File: solutions/java/logs-logs-logs/1/src/main/java/LogLevel.java
public enum LogLevel {
    UNKNOWN(0),
    TRACE(1),
    DEBUG(2),
    INFO(4),
    WARNING(5),
    ERROR(6),
    FATAL(42);

    private final int logLevel;

    LogLevel(int logLevel) {
        this.logLevel = logLevel;
    }

    public int getLogLevelNumbers() {
        return logLevel;
    }

    public String getLogLevelString() {
        return switch (logLevel) {
            case 0 -> "UKW";
            case 1 -> "TRC";
            case 2 -> "DBG";
            case 4 -> "INF";
            case 5 -> "WRN";
            case 6 -> "ERR";
            case 42 -> "FTL";
            default -> "XYZ";
        };
    }

    public static LogLevel getLogLevelFromString(String logLevelString) {
        return switch (logLevelString) {
            case "TRC" -> LogLevel.TRACE;
            case "DBG" -> LogLevel.DEBUG;
            case "INF" -> LogLevel.INFO;
            case "WRN" -> LogLevel.WARNING;
            case "ERR" -> LogLevel.ERROR;
            case "FTL" -> LogLevel.FATAL;
            default -> LogLevel.UNKNOWN;
        };
    }
}
